algopy_testing/src/algopy/primitives/uint64.py

```diff
@@ -0,0 +1,735 @@
+        return UInt64(other // self.value)
+
+    def __ifloordiv__(self, other: int | UInt64) -> Self:
+        """
+        In-place floor division of the UInt64 with another UInt64 or int.
+
+        Args:
+            other (int | UInt64): The object to divide the UInt64 by.
+
+        Raises:
+            UInt64ZeroDivisionError: If the divisor is zero.
+
+        Returns:
+            Self: The result of the in-place floor division.
+        """
+        other_value = self._get_value(other)
+        if other_value == 0:
+            raise UInt64ZeroDivisionError(other_value)
+        self.value //= other_value
+        return self
+
+    def __mod__(self, other: int | UInt64) -> UInt64:
+        """
+        Compute the modulus of the UInt64 by another UInt64 or int.
+
+        Args:
+            other (int | UInt64): The object to compute the modulus by.
+
+        Raises:
+            UInt64ZeroDivisionError: If the divisor is zero.
+
+        Returns:
+            UInt64: The result of the modulus operation.
+        """
+        other_value = self._get_value(other)
+        if other_value == 0:
+            raise UInt64ZeroDivisionError(other_value)
+        return UInt64(self.value % other_value)
+
+    def __rmod__(self, other: int) -> UInt64:
+        """
+        Compute the modulus of an int by the UInt64.
+
+        Args:
+            other (int): The int to compute the modulus by the UInt64.
+
+        Raises:
+            UInt64ZeroDivisionError: If the UInt64 is zero.
+
+        Returns:
+            UInt64: The result of the modulus operation.
+        """
+        if self.value == 0:
+            raise UInt64ZeroDivisionError(self.value)
+        return UInt64(other % self.value)
+
+    def __imod__(self, other: int | UInt64) -> Self:
+        """
+        In-place modulus of the UInt64 with another UInt64 or int.
+
+        Args:
+            other (int | UInt64): The object to compute the modulus by.
+
+        Raises:
+            UInt64ZeroDivisionError: If the divisor is zero.
+
+        Returns:
+            Self: The result of the in-place modulus operation.
+        """
+        other_value = self._get_value(other)
+        if other_value == 0:
+            raise UInt64ZeroDivisionError(other_value)
+        self.value %= other_value
+        return self
+
+    def __pow__(self, other: int | UInt64, modulo: int | UInt64 | None = None) -> UInt64:
+        """
+        Raise the UInt64 to the power of another UInt64 or int.
+
+        Args:
+            other (int | UInt64): The exponent to raise the UInt64 to.
+            modulo (int | UInt64 | None, optional): The modulus to apply to the result.
+            Defaults to None.
+
+        Raises:
+            UInt64OverflowError: If the power result exceeds the maximum value for a UInt64.
+
+        Returns:
+            UInt64: The result of the power operation.
+        """
+        exp = self._get_value(other)
+        if modulo is not None:
```

Comment:
we don't support the `pow` function yet

Reply:
For now am returning `NotImplemented` if a modulo is specified